src/OpenLoco/src/Objects/ObjectManager.cpp

```diff
@@ -412,22 +605,22 @@ namespace OpenLoco::ObjectManager
         }
 
         // Copy the object into Loco freeable memory (required for when load loads the object)
-        preLoadObj.object = reinterpret_cast<Object*>(malloc(data.size()));
+        preLoadObj.object = std::make_unique<std::byte[]>(data.size());
         if (preLoadObj.object == nullptr)
         {
             return std::nullopt;
         }
-        std::copy(std::begin(data), std::end(data), reinterpret_cast<std::byte*>(preLoadObj.object));
+        std::copy(std::begin(data), std::end(data), preLoadObj.object.get());
 
-        preLoadObj.objectData = std::span<std::byte>(reinterpret_cast<std::byte*>(preLoadObj.object), data.size());
+        preLoadObj.objectData = std::span<std::byte>(preLoadObj.object.get(), data.size());
 
-        if (!callObjectValidate(preLoadObj.header.getType(), *preLoadObj.object))
-        {
-            free(preLoadObj.object);
-            // Object failed validation
-            Logging::error("Object {} in index failed validation! (This should not be possible)", header.getName());
-            return std::nullopt;
-        }
+        // if (!callObjectValidate(preLoadObj.header.getType(), *preLoadObj.object))
```

Comment:
Need to work out a way of doing this. Should we make the field by field reader a seperate member function so we can use it during validate?

Reply:
Should be safe to move this.